### PR TITLE
fix(hubspot): update URL and method to get result + remove legacy mode

### PR DESCRIPTION
**Pull request by f-bb-toucantoco**
## Change Summary
Behavior of change: Hubspot works

- Add mandatory scope in hubspot/doc.md
- Update all route 
- Remove 2 hubspot endpoint temporary : web-analytics && emails-events
- Remove legacy mode

## Checklist

* [X] Unit tests for the changes exist
* [X] Tests pass on CI and coverage remains at 100%
* [X] Documentation reflects the changes where applicable

### CODE REVIEW

**Comment by f-bb-toucantoco** [comment]
[Bug - 2- As an AM, I'd like to have Hubspot connector work :)](https://trello.com/c/HTmyNdAj/923-bug-2-as-an-am-id-like-to-have-hubspot-connector-work)

File: tests/hubspot/test_hubspot.py [hunk 1]
```diff
@@ -92,99 +78,17 @@ def test_hubspot_wrong_data_return(connector, datasource, mocker):
...
-    try:
+    with pytest.raises(HubspotConnectorException):
         connector.get_df(datasource)
-    except Exception as e:
-        assert isinstance(e, HubspotConnectorException)
-
-    assert req_mock.called_once()
-
-
-def test_hubspot_get_webanalytics(connector, datasource, mocker):
```
Comment (Sanix-Darker) [inline tests/hubspot/test_hubspot.py:103]: Is this test not important ?
It looks important for me... 🤔

File: toucan_connectors/hubspot/hubspot_connector.py [hunk 2]
```diff
@@ -102,37 +113,36 @@ def build_authorization_url(self, **kwargs):
     def _get_access_token(self):
         return self.__dict__['_oauth2_connector'].get_access_token()
 
-    def _handle_pagination(self, endpoint_info, query_params, headers):
-        url = endpoint_info['url']
-        legacy = endpoint_info['legacy']
+    def _handle_pagination(
+        self, endpoint_info: Dict[str, Any], data_name: str, query_params: Dict[str, str], headers
```
Comment (Sanix-Darker) [inline toucan_connectors/hubspot/hubspot_connector.py:117]: i may be wrong, but the `data_name` you're calling in this method should not be the `dataset` ? based on the method `_retrieve_data` at line 155... 🤔

File: tests/hubspot/test_hubspot.py (hunk 1 above)
Comment (f-bb-toucantoco) [reply tests/hubspot/test_hubspot.py:103]: I have removed the part on webanalytics route
So this test is not necessary

**Review by Sanix-Darker — APPROVED** [review]

**Pull request merged**